src/Microsoft.AspNetCore.Razor.Language/Legacy/RazorParser.cs

```diff
@@ -37,7 +37,10 @@ public virtual RazorSyntaxTree Parse(RazorSourceDocument source)
 
             var reader = new SeekableTextReader(chars, source.FileName);
 
-            var context = new ParserContext(reader, Options.DesignTimeMode);
+            var context = new ParserContext(reader, Options.DesignTimeMode)
+            {
+                StopParsingAfterFirstDirective = Options.StopParsingAfterFirstDirective
+            };
```

Comment:
Hmm, we wouldn't add a parse single directive method to RazorSyntaxTree correct? Just trying to validate that we don't add noise to currently super simple APIs.

Currently the options approach is a somewhat hidden approach that MVC can utilize and not drastically impact other consumers.